app/src/app/api/v1/mcp/route.ts

```diff
@@ -0,0 +1,383 @@
+ *                 type: string
+ *                 enum: ["2.0"]
+ *               method:
+ *                 type: string
+ *                 enum: ["initialize", "initialized", "tools/list", "tools/call", "ping"]
+ *               params:
+ *                 type: object
+ *               id:
+ *                 oneOf:
+ *                   - type: string
+ *                   - type: number
+ *                   - type: "null"
+ *     responses:
+ *       200:
+ *         description: MCP response
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 jsonrpc:
+ *                   type: string
+ *                 result:
+ *                   type: object
+ *                 error:
+ *                   type: object
+ *                 id:
+ *                   oneOf:
+ *                     - type: string
+ *                     - type: number
+ *                     - type: "null"
+ *       401:
+ *         description: Authentication required for tool calls
+ *       500:
+ *         description: Internal server error
+ */
+import { apiHandler } from "@/util/api";
+import { NextRequest, NextResponse } from "next/server";
+import { logger } from "@/services/logger";
+import { McpError, ErrorCode, Tool } from "@modelcontextprotocol/sdk/types.js";
+import { AppSession } from "@/lib/auth";
+
+// Import tool implementations
+import * as inventoriesTools from "@/lib/mcp/tools/inventories";
+import * as emissionsTools from "@/lib/mcp/tools/emissions";
+import * as citiesTools from "@/lib/mcp/tools/cities";
+import * as cityProfileTools from "@/lib/mcp/tools/city-profile";
+import * as actionPlansTools from "@/lib/mcp/tools/action-plans";
+import * as riskAssessmentTools from "@/lib/mcp/tools/risk-assessment";
+
+// Tool registry
+const toolRegistry = new Map<string, { definition: Tool; handler: any }>();
+
+// Register all tools
+function registerTools() {
+  // Inventory tools
+  toolRegistry.set("get_user_inventories", {
+    definition: inventoriesTools.getUserInventoriesTool,
+    handler: inventoriesTools,
+  });
+
+  toolRegistry.set("get_inventory_emissions", {
+    definition: emissionsTools.getInventoryEmissionsTool,
+    handler: emissionsTools,
+  });
+
+  // City tools
+  toolRegistry.set("get_user_cities", {
+    definition: citiesTools.getUserCitiesTool,
+    handler: citiesTools,
+  });
+
+  toolRegistry.set("get_city_profile", {
+    definition: cityProfileTools.getCityProfileTool,
+    handler: cityProfileTools,
+  });
+
+  // Strategy tools
+  toolRegistry.set("get_climate_action_plans", {
+    definition: actionPlansTools.getClimateActionPlansTool,
+    handler: actionPlansTools,
+  });
+
+  toolRegistry.set("get_climate_risk_assessment", {
+    definition: riskAssessmentTools.getClimateRiskAssessmentTool,
+    handler: riskAssessmentTools,
+  });
+}
+
+// Initialize tools on module load
+registerTools();
```

Comment:
Would be nicer to have an init method (so this is only called when used, and maybe as a one-time init effect that is ignored for the next call), but not super urgent.